sdk/storage/azure-storage-blob/swagger/README.md

```diff
@@ -0,0 +1,849 @@
+    }
+```
+
+### /{containerName}/{blob}?comp=lease&renew
+``` yaml
+directive:
+- from: swagger-document
+  where: $["x-ms-paths"]["/{containerName}/{blob}?comp=lease&renew"]
+  transform: >
+    let param = $.put.parameters[0];
+    if (!param["$ref"].endsWith("ContainerName")) {
+      const path = param["$ref"].replace(/[#].*$/, "#/parameters/");
+      $.put.parameters.splice(0, 0, { "$ref": path + "ContainerName" });
+      $.put.parameters.splice(1, 0, { "$ref": path + "Blob" });
+    }
+```
+
+### /{containerName}/{blob}?comp=lease&change
+``` yaml
+directive:
+- from: swagger-document
+  where: $["x-ms-paths"]["/{containerName}/{blob}?comp=lease&change"]
+  transform: >
+    let param = $.put.parameters[0];
+    if (!param["$ref"].endsWith("ContainerName")) {
+      const path = param["$ref"].replace(/[#].*$/, "#/parameters/");
+      $.put.parameters.splice(0, 0, { "$ref": path + "ContainerName" });
+      $.put.parameters.splice(1, 0, { "$ref": path + "Blob" });
+    }
+```
+
+### /{containerName}/{blob}?comp=lease&break
+``` yaml
+directive:
+- from: swagger-document
+  where: $["x-ms-paths"]["/{containerName}/{blob}?comp=lease&break"]
+  transform: >
+    let param = $.put.parameters[0];
+    if (!param["$ref"].endsWith("ContainerName")) {
+      const path = param["$ref"].replace(/[#].*$/, "#/parameters/");
+      $.put.parameters.splice(0, 0, { "$ref": path + "ContainerName" });
+      $.put.parameters.splice(1, 0, { "$ref": path + "Blob" });
+    }
+```
+
+### /{containerName}/{blob}?comp=snapshot
+``` yaml
+directive:
+- from: swagger-document
+  where: $["x-ms-paths"]["/{containerName}/{blob}?comp=snapshot"]
+  transform: >
+    let param = $.put.parameters[0];
+    if (!param["$ref"].endsWith("ContainerName")) {
+      const path = param["$ref"].replace(/[#].*$/, "#/parameters/");
+      $.put.parameters.splice(0, 0, { "$ref": path + "ContainerName" });
+      $.put.parameters.splice(1, 0, { "$ref": path + "Blob" });
+    }
+```
+
+### /{containerName}/{blob}?comp=copy
+``` yaml
+directive:
+- from: swagger-document
+  where: $["x-ms-paths"]["/{containerName}/{blob}?comp=copy"]
+  transform: >
+    let param = $.put.parameters[0];
+    if (!param["$ref"].endsWith("ContainerName")) {
+      const path = param["$ref"].replace(/[#].*$/, "#/parameters/");
+      $.put.parameters.splice(0, 0, { "$ref": path + "ContainerName" });
+      $.put.parameters.splice(1, 0, { "$ref": path + "Blob" });
+      $.put.responses["202"].headers["x-ms-version-id"] = {
+        "x-ms-client-name": "VersionId",
+        "type": "string",
+        "description": "UTC date/time value generated by the service that identifies the version of the blob. This header is returned for requests made against version 2018-11-09 and above."
+      };
+    }
+```
+
+### /{containerName}/{blob}?comp=copy&sync
+``` yaml
+directive:
+- from: swagger-document
+  where: $["x-ms-paths"]["/{containerName}/{blob}?comp=copy&sync"]
```

Comment:
Can't this be combined with the comp=copy version directly above this?

Reply:
I'm not aware of a way to do that.

Reply:
I thought the where clauses were json path syntax and if so I'd expect something like `/{containerName}/{blob}?comp=copy*` to work. Although I'm no expert in json path or these autorest directives so perhaps that is just wishful thinking.

Reply:
Maybe?  [The only documentation I've found](https://github.com/Azure/autorest/tree/master/Samples/3b-custom-transformations) doesn't mention that explicitly, but it does look like JSON Path syntax.  I don't think you can use a wildcard `*` inside a string like that, but there's probably a way using those arcane JSON Path filter expressions.  I don't want to block progress on getting this work done though so we can explore that after this PR.